codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/customizations/HttpConnectorConfigDecorator.kt

```diff
@@ -183,7 +188,29 @@ private class HttpConnectorConfigCustomization(
             }
 
             is ServiceConfig.BuilderBuild -> writable {
-                if (runtimeMode.defaultToMiddleware) {
+                if (runtimeMode.defaultToOrchestrator) {
+                    rustTemplate(
+                        """
+                        let sleep_impl = layer.load::<#{SharedAsyncSleep}>().cloned();
+                        let timeout_config = layer.load::<#{TimeoutConfig}>().cloned().unwrap_or_else(|| #{TimeoutConfig}::disabled());
```

Comment:
So if we set a `TimeoutConfig` on the top-level config, and then change the `http_connector` in a `config_override`, will it use the top-level `TimeoutConfig` or `TimeoutConfig::disabled()`?

Reply:
The implementation of `config_override` has been [under construction](https://github.com/awslabs/smithy-rs/blob/main/codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/config/ServiceConfigGenerator.kt#L459) (and still is as of this PR), so I'm speculating from my mental model. I think it uses the top-level `TimeoutConfig`.

Since whatever is set through `config_override` is applied to the operation-level configuration, as opposed to the top-level service configuration, during the execution of the orchestrator, an operation runtime plugin will have access to a config bag that has accumulated layers so far (among which is a frozen layer from the top-level service config). So it _should_ use the top-level `TimeoutConfig` rather than `TimeoutConfig::disabled()`.